src/rdb.c

```diff
@@ -2183,6 +2231,205 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
 
         /* All pairs should be read by now */
         serverAssert(len == 0);
+    } else if (rdbtype == RDB_TYPE_HASH_METADATA) {
+
+        size_t fieldLen;
+        sds value, field;
+        uint64_t ttl, minExpire = UINT64_MAX;
+        mstime_t now = mstime();
+        dict *dupSearchDict = NULL;
+        void *hashAddCtx = NULL;
+
+        len = rdbLoadLen(rdb, NULL);
+        if (len == RDB_LENERR) return NULL;
+        if (len == 0) goto emptykey;
+
+        redisDebug("loading metada-enriched hash with %lu keys", len);
+
+        o = createHashObject();
+        /* Too many entries? Use a hash table right from the start. */
+        if (len > server.hash_max_listpack_entries) {
+            hashTypeConvert(db, o, OBJ_ENCODING_HT);
+            hashAddCtx = HashTypeGroupSetInit(key, o, db);
```

Comment:
i don't like using hashAddCtx as this way, did you just for updating the ebuckets in the hashTypeSetExDone()?
if so, i think we can extract these code to separate method.

Reply:
This is the (short version) of the new API opened by @moticless in t_hash to add fields to a hash object, plus setting their expiration time.
Not commenting on the specific implementation, I think it is better (in terms of encapsulation, at least) to have the logic of adding / deleting / setting expiry (of) fields in t_hash.c instead of rdb.c.
Is there any other specific suggestion?

Reply:
it can be complicated, i think we can leave it as now, and improve it in the future.